DatadogFlags/Sources/FlagsClient.swift

```diff
@@ -53,84 +78,82 @@ public class FlagsClient {
-                        }
+                        let response = try JSONDecoder().decode(FlagAssignmentsResponse.self, from: data)
+                        self.store.setFlagAssignments(response.flags, for: context, date: dateProvider.now)
+                        completion(.success(()))
                     } catch {
-                        completion(.failure(.networkError(error)))
+                        completion(.failure(.invalidResponse))
                     }
-
                 case .failure(let error):
                     completion(.failure(.networkError(error)))
                 }
             }
         }
     }
 
+    @inlinable
     public func getBooleanValue(key: String, defaultValue: Bool) -> Bool {
-        let flags = store.getFlags()
-        if let flagData = flags[key] as? [String: Any],
-           let value = flagData["variationValue"] as? Bool {
-            return value
-        }
-        return defaultValue
+        getValue(key: key, defaultValue: defaultValue)
     }
 
+    @inlinable
     public func getStringValue(key: String, defaultValue: String) -> String {
-        let flags = store.getFlags()
-        if let flagData = flags[key] as? [String: Any],
-           let value = flagData["variationValue"] as? String {
-            return value
-        }
-        return defaultValue
+        getValue(key: key, defaultValue: defaultValue)
     }
 
-    public func getIntegerValue(key: String, defaultValue: Int64) -> Int64 {
-        let flags = store.getFlags()
-        if let flagData = flags[key] as? [String: Any],
-           let value = flagData["variationValue"] as? NSNumber {
-            return value.int64Value
-        }
-        return defaultValue
+    @inlinable
+    public func getIntegerValue(key: String, defaultValue: Int) -> Int {
+        getValue(key: key, defaultValue: defaultValue)
     }
 
+    @inlinable
     public func getDoubleValue(key: String, defaultValue: Double) -> Double {
-        let flags = store.getFlags()
-        if let flagData = flags[key] as? [String: Any],
-           let value = flagData["variationValue"] as? NSNumber {
-            return value.doubleValue
-        }
-        return defaultValue
+        getValue(key: key, defaultValue: defaultValue)
     }
 
-    // TODO: FFL-1047 Replace [String: Any] with OpenFeature.Value-compatible type
-    public func getObjectValue(key: String, defaultValue: [String: Any]) -> [String: Any] {
-        let flags = store.getFlags()
-        if let flagData = flags[key] as? [String: Any],
-           let value = flagData["variationValue"] as? [String: Any] {
-            return value
+    @inlinable
+    public func getObjectValue(key: String, defaultValue: AnyValue) -> AnyValue {
+        getValue(key: key, defaultValue: defaultValue)
+    }
+
+    @inlinable
+    public func getObjectValue<T: Decodable>(key: String, defaultValue: T, using decoder: JSONDecoder = .init()) -> T {
```

Comment:
question/ Isn't `decoder: JSONDecoder = .init()` allocating new JSONDecoder each time? That might be expensive, no?

Reply:
🤔 I don't think we can use a static `JSONDecoder` instance as a fallback. As far as I know, `JSONDecoder` is not thread-safe. So, unless we create a static wrapper with a lock, the safest approach is to create a new instance. The time it takes to create a `JSONDecoder` instance is negligible, compared to the time it takes decoding a JSON.

We still have the option to remove this API completely. In the end it is only a nice convenience and not required by our product.